CDLUC3/dmsp_frontend_prototype
# [Feature] - Update Edit Template page, including Publish functionality, and hook up to backend

## Description

- Updated the Edit Template page to get data from the backend. You need the backend updates ([#153](https://github.com/CDLUC3/dmsp_backend_prototype/issues/153)) to test with the frontend changes.
- Added translations
- Added unit tests
- Updated Question styles to only show 3 lines of the question, with an ellipsis
- Added mutations for createTemplateVersion and archiveTemplate
- Added query for templateQuery

Fixes # ([206](https://github.com/CDLUC3/dmsp_frontend_prototype/issues/206))

## Type of change

## How Has This Been Tested?
This was tested manually, as well as with unit tests.

To test,
- Get `feature/JS-add-lastPublishedDate-to-templates-table` branch of dmsp_backend_prototype
- Start this frontend branch
- Log in with `super@example.com` and navigate to `/en-US/template/15`
- Note that the Sections and questions display
- Note that when you click "Publish Template" button, you can save a versionedTemplate with the specified "visibility" and "changelog"
- Note that when you click "Save Draft", that a versionedTemplate is created with a versionType of "Draft"


## Checklist:
- [ ] Any dependent changes have been merged and published in downstream modules (the dependent Backend changes need to be merged first)
